mitosheet/src/mito/components/taskpanes/Graph/GraphSidebar.tsx

```diff
@@ -1,48 +1,66 @@
-import React, { useEffect, useState } from 'react';
-import { MitoAPI,  getRandomId } from '../../../api/api';
-import { useDebouncedEffect } from '../../../hooks/useDebouncedEffect';
-import { useEffectOnUpdateEvent } from '../../../hooks/useEffectOnUpdateEvent';
-import { AnalysisData, ColumnID, ColumnIDsMap, GraphDataDict, GraphID, GraphSidebarTab, SheetData, UIState } from '../../../types';
-import DefaultEmptyTaskpane from '../DefaultTaskpane/DefaultEmptyTaskpane';
-import { TaskpaneType } from '../taskpanes';
-import GraphSidebarTabs from './GraphSidebarTabs';
-import LoadingSpinner from './LoadingSpinner';
-import { getGraphParams } from './graphUtils';
-
-// import css
+import React, { useEffect } from 'react';
 import '../../../../../css/taskpanes/Graph/GraphSidebar.css';
 import '../../../../../css/taskpanes/Graph/LoadingSpinner.css';
-import Row from '../../layout/Row';
-import Col from '../../layout/Col';
+import { MitoAPI } from '../../../api/api';
+import { useEffectOnResizeElement } from '../../../hooks/useEffectOnElementResize';
+import useLiveUpdatingParams from '../../../hooks/useLiveUpdatingParams';
+import { AnalysisData, ColumnID, GraphDataArray, GraphID, GraphParamsBackend, GraphParamsFrontend, GraphSidebarTab, SheetData, StepType, UIState } from '../../../types';
 import XIcon from '../../icons/XIcon';
-import GraphStyleTab from './GraphStyleTab';
-import GraphSetupTab from './GraphSetupTab';
+import Col from '../../layout/Col';
+import Row from '../../layout/Row';
+import DefaultEmptyTaskpane from '../DefaultTaskpane/DefaultEmptyTaskpane';
+import { TaskpaneType } from '../taskpanes';
 import GraphExportTab from './GraphExportTab';
-import { useEffectOnResizeElement } from '../../../hooks/useEffectOnElementResize';
+import GraphSetupTab, { GraphType } from './GraphSetupTab';
+import GraphSidebarTabs from './GraphSidebarTabs';
+import GraphStyleTab from './GraphStyleTab';
+import LoadingSpinner from './LoadingSpinner';
+import { convertBackendtoFrontendGraphParams, convertFrontendtoBackendGraphParams, getDefaultGraphParams, getGraphRenderingParams, getParamsForExistingGraph } from './graphUtils';
 
+export type OpenGraphType = {
+    type: 'existing_graph'
+    graphID: GraphID,
+    existingParams: GraphParamsFrontend
+} | {
+    type: 'new_graph'
+    graphID: GraphID,
+    graphType: GraphType
+    selectedColumnIds?: ColumnID[]
+} | {
+    type: 'new_graph_duplicated_from_existing',
+    graphID: GraphID,
+    graphIDOfDuplicated: GraphID,
```

Comment:
newGraphId

Reply:
graphID is good because then we can use graphID on the openGraph no matter what the type is -- making a lot of code simpler